cpp/src/arrow/ipc/memory.cc

```diff
@@ -144,5 +144,29 @@ Status MemoryMappedSource::Write(int64_t position, const uint8_t* data, int64_t
   return Status::OK();
 }
 
+MockMemorySource::MockMemorySource(int64_t size) : size_(size) {}
```

Comment:
Per @emkornfield note below, we don't want to put unnecessary compiled symbols in the library object code